### PR TITLE
Prompt warns when game ought to be ended to due fifty-moves rules

### PR DESCRIPTION
**Pull request by AlexisMlz**

### CODE REVIEW

File: sources/engine/ChessEngine.cpp [hunk 1]
```diff
@@ -92,19 +92,24 @@ void    ChessEngine::move(Piece *piece, Piece::pos2d pos)
     if (m->captured) {
         pieces.remove(m->captured);
         delete m->captured;
+        fmCounter=0;
```
Comment (Sacrimento) [inline sources/engine/ChessEngine.cpp]: spaces

File: sources/engine/ChessEngine.cpp [hunk 2]
```diff
@@ -92,19 +92,24 @@ void    ChessEngine::move(Piece *piece, Piece::pos2d pos)
...
     }
 
-    if (m->flag != Move::Flag::Promotion)
+    if (m->flag != Move::Flag::Promotion){
```
Comment (Sacrimento) [inline sources/engine/ChessEngine.cpp]: space

File: sources/engine/ChessEngine.cpp [hunk 3]
```diff
@@ -92,19 +92,24 @@ void    ChessEngine::move(Piece *piece, Piece::pos2d pos)
...
         piece->move(pos);
+        fmCounter = piece->getType()!=Piece::Type::PAWN ? ++fmCounter : 0;
```
Comment (Sacrimento) [inline sources/engine/ChessEngine.cpp]: spaces

Comment (Sacrimento) [reply sources/engine/ChessEngine.cpp]: Also `fmCounter + 1` would be clearer than `++fmCounter` imo

File: sources/engine/ChessEngine.cpp [hunk 4]
```diff
@@ -92,19 +92,24 @@ void    ChessEngine::move(Piece *piece, Piece::pos2d pos)
...
+    }
     else
     {
         pieces.remove(m->moving);
         pieces.push_back(createPiece(promotionType, m->moving->getColor(), m->pos));
         delete m->moving;
+        fmCounter=0;
```
Comment (Sacrimento) [inline sources/engine/ChessEngine.cpp]: spaces

File: sources/engine/ChessEngine.h [hunk 5]
```diff
@@ -56,6 +56,8 @@ class ChessEngine
         Piece::Type     promotionType = Piece::Type::QUEEN;
         Piece::Color    turn;
 
+        int fmCounter; //"fifty-moves"
```
Comment (Sacrimento) [inline sources/engine/ChessEngine.h]: It always will be unsigned, so `uint8_t` ?

**Review by Sacrimento — CHANGES_REQUESTED** [review]

File: sources/engine/ChessEngine.h (hunk 5 above)
Comment (Greesb) [reply sources/engine/ChessEngine.h]: `moveCounter` is better than `fmCounter` imo

Comment (AlexisMlz) [reply sources/engine/ChessEngine.h]: Based on fen format, `moveCounter` should be used to describe another variable.

**Review by Sacrimento — DISMISSED** [review]

**Pull request merged**